components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/RequestObjectValidatorImpl.java

```diff
@@ -0,0 +1,331 @@
+    private static final String JWK = "jwk";
+    private static final String X5U = "x5u";
+    private static final String X5C = "x5c";
+    private static final String X5T_S256 = "x5t#s256";
+    //JWE is consists of five parts seperated by 4 '.'s as JOSE header , JWE encrypted key, Initialization vector,
+    // Cipher text and Authentication Tag
+    private static final int NUMBER_OF_PARTS_IN_JWE = 5;
+    //JWS is consists of three parts seperated by 2 '.'s as JOSE header, JWS payload, JWS signature
+    private static final int NUMBER_OF_PARTS_IN_JWS = 3;
+
+    private static Log log = LogFactory.getLog(RequestObjectValidatorImpl.class);
+    private static Properties properties;
+    private String jwtAssertion;
+    private byte[] jwtSignature;
+    private String headerValue;
+    private static String payload;
+    private static final Base64 base64Url = new Base64(true);
+
+    public String getPayload() {
+        return payload;
+    }
+
+    public void setPayload(String payload) {
+        RequestObjectValidatorImpl.payload = payload;
+    }
+
+    @Override
+    public void validateSignature(String requestObject) throws RequestObjectException {
+        String thumbPrint;
+        // The public key corresponding to the key used to sign the message can be any of these header elements:
+        // jku, jwk, kid, x5u, x5c, x5t and x5t#s256
+        if (getJsonHeaderObject() != null) {
+            if (getJsonHeaderObject().get(X5T) != null) {
+                thumbPrint = getJsonHeaderObject().get(X5T).toString();
+            } else if (getJsonHeaderObject().get(KID) != null) {
+                thumbPrint = getJsonHeaderObject().get(KID).toString();
+            } else if (getJsonHeaderObject().get(JKU) != null) {
+                thumbPrint = getJsonHeaderObject().get(JKU).toString();
+            } else if (getJsonHeaderObject().get(JWK) != null) {
+                thumbPrint = getJsonHeaderObject().get(JWK).toString();
+            } else if (getJsonHeaderObject().get(X5U) != null) {
+                thumbPrint = getJsonHeaderObject().get(X5U).toString();
+            } else if (getJsonHeaderObject().get(X5C) != null) {
+                thumbPrint = getJsonHeaderObject().get(X5C).toString();
+            } else if (getJsonHeaderObject().get(X5T_S256) != null) {
+                thumbPrint = getJsonHeaderObject().get(X5T_S256).toString();
+            } else {
+                throw new RequestObjectException(RequestObjectException.ERROR_CODE_INVALID_REQUEST, "Can not find the " +
+                        "certificate thumbprint for signature validation");
+            }
+        } else {
+            throw new RequestObjectException(RequestObjectException.ERROR_CODE_INVALID_REQUEST, "The request object " +
+                    "header is null. Hence signature validation failed.");
+        }
+        if (log.isDebugEnabled()) {
+            log.debug("The certificate thumbPrint value for the certificate is: " + thumbPrint);
```

Comment:
how about logging the header that came in the request that helped identify the thumbprint